Assets/Photon Unity Networking/UtilityScripts/PunTeams.cs

```diff
@@ -102,7 +102,7 @@ public static PunTeams.Team GetTeam(this PhotonPlayer player)
         object teamId;
         if (player.CustomProperties.TryGetValue(PunTeams.TeamPlayerProp, out teamId))
         {
-            return (PunTeams.Team)teamId;
+            return (PunTeams.Team)(int)teamId;
```

Comment:
Is there a reason why this change is introduced? Photon Unity Networking is an unity asset (third party library). We shouldn't modify this.

Reply:
L56 in ChatUtility.cs, the player.GetTeam() method I assumed should return a PunTeams.Team enum, but it didnt until I made the cast in that method, but I see now that I maybe should make the cast outside of method instead. I'll change it.

Reply:
InvalidCastException: Specified cast is not valid.
TeamExtensions.GetTeam (PhotonPlayer player) (at Assets/Photon Unity Networking/UtilityScripts/PunTeams.cs:105)
ChatUtility.SetNameColorDependingOnteam (PhotonPlayer player) (at Assets/Scripts/Room/Chat/ChatUtility.cs:56)
InRoomChat.ChatAll (System.String message) (at Assets/Scripts/Room/Chat/InRoomChat.cs:77)
InRoomChat.HandleChatInput (InRoomChat chat) (at Assets/Scripts/Room/Chat/InRoomChat.cs:116)
InRoomChat.OnGUI () (at Assets/Scripts/Room/Chat/InRoomChat.cs:30)

Reply:
It seems that casting object to PunTeams.Team is not valid for some reason

Reply:
Let's see. Perhaps the problem is that the PunTeams.Team is of type byte and not of type integer?

Reply:
Yeah but in my switch case I am using Punteams.Team

`var playerTeam = player.GetTeam();
        switch (playerTeam)
        {
            case PunTeams.Team.red:
                name = FormatTextColor00FFFF(name);
                break;
            case PunTeams.Team.blue:
                name = FormatTextColorFF00FF(name);
                break;
            default:
                break;
        }
`